src/js/media-chrome-gesture.js

```diff
@@ -0,0 +1,124 @@
+import { MediaUIAttributes, PointerTypes } from './constants.js';
+import { defineCustomElement } from './utils/defineCustomElement.js';
+import {
+  Window as window,
+  Document as document,
+} from './utils/server-safe-globals.js';
+
+const template = document.createElement('template');
+
+template.innerHTML = `
+<style>
+  :host {
+    display: inline-block;
+    width: auto;
+    height: auto;
+    box-sizing: border-box;
+
+    pointer-events: auto;
+  }
+</style>
+`;
+
+const allPointerTypes = Object.values(PointerTypes);
+
+class MediaChromeGesture extends window.HTMLElement {
+  static get observedAttributes() {
+    return [MediaUIAttributes.MEDIA_CONTROLLER, 'pointer-types'];
+  }
+
+  constructor(options = {}) {
+    super();
+
+    const shadow = this.attachShadow({ mode: 'open' });
+
+    const buttonHTML = template.content.cloneNode(true);
+    this.nativeEl = buttonHTML;
+
+    // Slots
+    let slotTemplate = options.slotTemplate;
+
+    if (!slotTemplate) {
+      slotTemplate = document.createElement('template');
+      slotTemplate.innerHTML = `<slot>${options.defaultContent || ''}</slot>`;
+    }
+
+    this.nativeEl.appendChild(slotTemplate.content.cloneNode(true));
+
+    shadow.appendChild(buttonHTML);
+
+    this._pointerType = undefined;
+    const pointerDownHandler = (e) => {
+      // Since not all browsers have updated to be spec compliant, where 'click' events should be PointerEvents,
+      // we can use use 'pointerdown' to reliably determine the pointer type. (CJP).
+      this._pointerType = e.pointerType;
+    };
+
+    this.addEventListener('pointerdown', pointerDownHandler);
+
+    this.addEventListener('click', (event) => {
```

Comment:
we could do a "fancier"/more complex version of this that uses the first `"pointerdown"`+`"click"` to determine if `"pointerdown"` is necessary, but I figured I wouldn't over complicate this as a first pass.